Azure/azure-cli
# [SQL] `az sql mi create`, `az sql mi update`: Add `--service-principal-type` parameter to support Win Auth (Kerberos)

**Description**
Adding Service Principal parameter to _az sql mi update_ and _az sql mi create_ command for Windows Authentication for MI feature.

**Testing Guide**
A scenario test has been added for update of MI with service principal.
For manual testing: use _az sql mi update_ and _az sql mi create_ with --service-principal-type SystemAssigned

This checklist is used to make sure that common guidelines for a pull request are followed.
